jetty-http/src/main/java/org/eclipse/jetty/http/HttpParser.java

```diff
@@ -1139,6 +1149,15 @@ protected boolean parseFields(ByteBuffer buffer)
                                 return _handler.messageComplete();
                             }
 
+                            // We found Transfer-Encoding headers, but none declared the 'chunked' token
+                            if (_hasTransferEncoding && _endOfContent != EndOfContent.CHUNKED_CONTENT)
+                            {
+                                // Assume Transfer-Encoding chunked per
+                                // https://tools.ietf.org/html/rfc7230#section-3.3.1
+                                _endOfContent = EndOfContent.CHUNKED_CONTENT;
```

Comment:
I don't think we should assume chunked.   Either it is a response that is EOF terminated, in which we allow the TEs  OR it is not, in which case it is an error.

Reply:
Fixed.